### ISSUE TITLE
[Enhancement] Migrate Xamarin.CommunityToolkit.UnitTests to NUnit

### ISSUE THREAD

**Issue by brminnick** [xamarin/XamarinCommunityToolkit#1083]
## Summary

This enhancement will migrate `Xamarin.CommunityToolkit.UnitTests` from xUnit (which it currently uses) to NUnit.

The reason for migrating to NUnit is two fold:
1. `Xamarin.CommunityToolkit.Markup.UnitTests` currently uses NUnit
  - Keeping the unit testing framework consistent will allow for a consistent experience for contributors and maintainers 
2. [xUnit doesn't support the command line command dotnet test for .NET Framework ](https://xunit.net/docs/getting-started/netcore/cmdline):
![](https://user-images.githubusercontent.com/13558917/111080287-fdb1bf00-84ba-11eb-951d-95289cb1f362.png)
  - Long story short, this is the bash script that is required for us to run .NET Framework tests for xUnit on macOS (that is now included in [`azure-pipelines.yml`](https://github.com/xamarin/XamarinCommunityToolkit/blob/5a688543c8d6fe27c15bcecdbd91eeada1f79b36/azure-pipelines.yml#L211):
```bash
mono "`find ~/.nuget/packages | grep net461 | grep xunit.console.exe | grep -v config`" "`find . -name Xamarin.CommunityToolkit.UnitTests.dll | grep bin | grep 461`"
```

## Who Will Do The Work?

- [x] I am willing to take this on myself

**Comment by brminnick** [comment]
Currently on-hold until we merge v1.1.0 to `main`

**Comment by PureWeen** [comment]
Just an FYI

We're migrating everything to xunit on the .NET MAUI side of things

**Comment by brminnick** [comment]
Thanks Shayne! Good to know! 

We'll be chatting about the Community Toolkit's .NET MAUI migration soon (we're still in the early planning stage for .NET MAUI support), and I'll make sure we factor this in 👍
